bin/tests/common.rs

```diff
@@ -481,3 +485,83 @@ pub async fn fetch_agent_metrics(
 
     Ok((status, body))
 }
+
+// The compiler/linter believes this function isn't used anywhere but it is currently
+// used in the retries and http integration tests. This flag disables that false positive.
+#[allow(dead_code)]
+fn stream_agent_metrics(
+    metrics_port: u16,
+    scrape_delay: Option<Duration>,
+) -> impl Stream<Item = Sample> {
+    stream::unfold(Vec::new(), move |state| async move {
+        let mut state = loop {
+            if !state.is_empty() {
+                break state;
+            }
+
+            // Hitting the metrics endpoint too quick may result in subsequent queries without
+            // any data points. Depending on need, one can specify a delay before scrapping giving
+            // the agent time to generate some new metric data.
+            if let Some(delay) = scrape_delay {
+                tokio::time::sleep(delay).await;
+            }
+
+            match fetch_agent_metrics(metrics_port).await {
+                Ok((StatusCode::OK, Some(body))) => {
+                    let body = body.lines().map(|l| Ok(l.to_string()));
+                    break Scrape::parse(body)
+                        .expect("failed to parse the metrics response")
+                        .samples;
+                }
+                Ok((status, _)) => panic!(
+                    "The /metrics endpoint returned status code {:?}, expected 200.",
+                    status
+                ),
+                Err(err) => panic!(
+                    "Failed to make HTTP request to metrics endpoint - reason: {:?}",
+                    err
+                ),
+            }
+        };
+
+        state.pop().map(|metric| (metric, state))
+    })
+}
+
+// The compiler/linter believes this function isn't used anywhere but it is currently
+// used in the retries and http integration tests. This flag disables that false positive.
+#[allow(dead_code)]
```

Comment:
Are these `dead_code` attributes used just for compiler/linter errors? Looks like it's complaining about functions and structs?

Reply:
Yes, the compiler/linter think this code is not used because the only place it is used is in some test code that is conditional to the integration_test flag. It's only really a concern for common code in the `bin/tests/*` modules.